ios/browser/api/ads/brave_ads.mm

```diff
@@ -147,6 +147,7 @@ - (instancetype)initWithStateStoragePath:(NSString*)path {
     adEventCache = new brave_ads::AdEventCache();
 
     adsClient = new AdsClientIOS(self);
+    adsClientNotifier = new brave_ads::AdsClientNotifier();
```

Comment:
`adsClientNotifier` should be created in `init` because we need to notify Brave Ads about  `applicationDidBecomeActive` event, which is fired before Brave Ads instance is created.
`adsClientNotifier` instance queues all the events and make it possible to fire them after ads are initialized.